integration/test/bookend_ems_test.go

```diff
@@ -31,52 +32,54 @@ var nodeScopedResolvingEmsList = []string{
 }
 
 func setupAlerts() {
-	totalAlerts := 0
+	numAlerts := 0
 	emsConfigDir := utils.GetHarvestRootDir() + "/conf/ems/9.6.0"
 	log.Info().Str("EmsConfigDir", emsConfigDir).Msg("Directory path")
 
 	// Fetch ems configured in template
 	_, issuingEmsNames, resolvingEmsNames = promAlerts.GetEmsAlerts(emsConfigDir, "ems.yaml")
 
 	// Identify supported issuing ems names for the given cluster
+	now := time.Now()
 	supportedIssuingEms = promAlerts.GenerateEvents(issuingEmsNames, nodeScopedIssuingEmsList)
-	log.Info().Msgf("Total supported issuing ems: %d", len(supportedIssuingEms))
+	log.Info().
+		Int("supportedIssuingEms", len(supportedIssuingEms)).
+		Str("dur", time.Since(now).Round(time.Millisecond).String()).
+		Msg("Supported issuing ems")
 
 	// Fetch previous prometheus alerts
 	oldAlertsData, _ = promAlerts.GetAlerts()
 
 	// Identify supported ems names for the given cluster
+	now = time.Now()
 	supportedResolvingEms = promAlerts.GenerateEvents(resolvingEmsNames, nodeScopedResolvingEmsList)
-	log.Info().Msgf("Total supported resolving ems:%d", len(supportedResolvingEms))
+	log.Info().
+		Int("supportedResolvingEms", len(supportedResolvingEms)).
+		Str("dur", time.Since(now).Round(time.Millisecond).String()).
+		Msg("Supported resolving ems")
 
 	// Fetch current prometheus alerts
-	newAlertsData, totalAlerts = promAlerts.GetAlerts()
-	if totalAlerts == 0 {
+	newAlertsData, numAlerts = promAlerts.GetAlerts()
+	if numAlerts == 0 {
 		log.Info().Msg("No alerts found in prometheus")
 	}
-	log.Info().Msgf("Total firing alerts %d", totalAlerts)
+	log.Info().Int("numAlerts", numAlerts).Msg("Firing alerts")
 }
 
 func TestEmsTestSuite(t *testing.T) {
 	utils.SkipIfMissing(t, utils.BookendEms)
 	setupAlerts()
 
 	// Evaluate bookend active ems events
-	foundBookendEms := make([]string, 0)
-
 	for _, issuingEms := range supportedIssuingEms {
-		// If the issuingEms did not exit before, then ignore the test-case.
+		// If the issuingEms did not exist before, then ignore the test-case.
 		if oldAlertsData[issuingEms] > 0 {
-			v := oldAlertsData[issuingEms] - newAlertsData[issuingEms]
-			if v < 1 {
-				foundBookendEms = append(foundBookendEms, issuingEms)
+			v := newAlertsData[issuingEms] - oldAlertsData[issuingEms]
+			if v > 1 {
```

Comment:
This would not be as per our test case.
If we have issuing ems(ex. volume.offline) count as 3[oldAlertsData] earlier and after we fire resolving ems(volume.online), the issuing ems count would be reduced to 2[newAlertsData]
So, v = 2-3 = -1

Condition would be,
```
if v >= 0 {
error log
}
```